src/Zalo/Provider.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace SocialiteProviders\Zalo;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'ZALO';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [''];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase('https://oauth.zaloapp.com/v3/auth', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getCodeFields($state = null)
+    {
+        $fields = [
+            'app_id' => $this->clientId,
+            'redirect_uri' => $this->redirectUrl,
+            'state' => $state,
+        ];
+        return array_merge($fields, $this->parameters);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAccessTokenResponse($code)
+    {
+        $response = $this->getHttpClient()->get($this->getTokenUrl(), [
+            'headers' => ['Accept' => 'application/json'],
+            'query' => $this->getTokenFields($code),
+        ]);
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://oauth.zaloapp.com/v3/access_token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get('https://graph.zalo.me/v2.0/me?access_token=' . $token . '&fields=id,birthday,name,gender,picture');
+
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id' => $user['id'],
+            'nickname' => null,
+            'name' => $user['name'],
+            // Emails are not officially supported in Zalo API
+            // This is a hypothetical email identified with the type "ID@zalo.me"
+            'email' => $user['id'] . '@zalo.me',
```

Comment:
can just remove this, it's not a required field

Reply:
check [SocialiteProviders/Providers/pull/371](https://github.com/SocialiteProviders/Providers/pull/371)

tks!